### DISCUSSION TITLE
File nesting in VSCode routes

### DISCUSSION THREAD

**Discussion by accuser** [sveltejs/kit#6068] [General]
I've just enabled File Nesting in VSCode, and thought I'd share. It is a feature that visually groups  related files together under a single parent file in the Explorer pane.

## Steps

To nest `+page.js` (or `+page.ts`) and `+page.server.js` (or `+page.server.ts`) under a `+page.svelte` file, if they exist:

1. Enable Explorer > File Nesting

2. Add the following in Explorer > File Nesting : Patterns

    Item: `+*.svelte`
    Value: `+${capture}.js, +${capture}.server.tjs, +${capture}.ts, +${capture}.server.ts`

This will also do the same for other files that start match the pattern `+*.svelte`, so will work with `+layout.svelte` files too.

**Comment by tbdrz** [comment]
Explorer > File Nesting can be found in `Settings -> Features -> Explorer -> File Nesting`

Or in `settings.json`:

```
"explorer.fileNesting.enabled": true
"explorer.fileNesting.patterns": {
    "+*.svelte": "+${capture}.js, +${capture}.server.tjs, +${capture}.ts, +${capture}.server.ts"
  }
```
